rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport.h

```diff
@@ -35,25 +252,555 @@ namespace rmw_fastrtps_cpp
+                            subros_message = align_(max_align, 0, subros_message, space);
+                        }
+                    }
+                    break;
+                default:
+                    printf("unknown type id %u\n", member->type_id_);
+                    throw std::runtime_error("unknown type");
+            }
+        }
+    }
+
+    return true;
+}
+
+template <typename MembersType>
+bool rmw_fastrtps_cpp::TypeSupport<MembersType>::deserializeROSmessage(
+    eprosima::fastcdr::Cdr &deser, const MembersType *members, void *ros_message, bool call_new)
+{
+    assert(members);
+    assert(ros_message);
+
+    for(unsigned long i = 0; i < members->member_count_; ++i)
+    {
+        const auto *member = members->members_ + i;
+        void *field = (char*)ros_message + member->offset_;
+
+        if(!member->is_array_)
+        {
+            switch(member->type_id_)
+            {
+                case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_BOOL:
+                    {
+                        deser >> *(bool*)field;
+                    }
+                    break;
+                case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_BYTE:
+                case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_UINT8:
+                    deser >> *(uint8_t*)field;
+                    break;
+                case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_CHAR:
+                case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_INT8:
+                    deser >> *(char*)field;
+                    break;
+                case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_FLOAT32:
+                    deser >> *(float*)field;
+                    break;
+                case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_FLOAT64:
+                    deser >> *(double*)field;
+                    break;
+                case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_INT16:
+                    deser >> *(int16_t*)field;
+                    break;
+                case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_UINT16:
+                    deser >> *(uint16_t*)field;
+                    break;
+                case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_INT32:
+                    deser >> *(int32_t*)field;
+                    break;
+                case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_UINT32:
+                    deser >> *(uint32_t*)field;
+                    break;
+                case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_INT64:
+                    deser >> *(int64_t*)field;
+                    break;
+                case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_UINT64:
+                    deser >> *(uint64_t*)field;
+                    break;
+                case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_STRING:
+                    {
+                        std::string &str = *(std::string*)field;
```

Comment:
I built rcl with this branch of rmw_fastrtps_cpp and removed the ignore logic in CMakeLists.txt for the tests, then ran `test_subscription__rmw_fastrtps_cpp` and got a segfault in the test case `test_subscription_nominal_string`:

```
0x00007ffff6d4b7e3 in std::string::swap(std::string&) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
(gdb) bt
#0  0x00007ffff6d4b7e3 in std::string::swap(std::string&) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#1  0x00007ffff6d4b829 in std::string::operator=(std::string&&) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#2  0x00007ffff6488eed in eprosima::fastcdr::Cdr::deserialize(std::string&) ()
   from /home/jackie/code/ros2_ws/install/lib/librmw_fastrtps_cpp.so
#3  0x00007ffff6488c09 in eprosima::fastcdr::Cdr::operator>>(std::string&) ()
   from /home/jackie/code/ros2_ws/install/lib/librmw_fastrtps_cpp.so
#4  0x00007ffff64958eb in rmw_fastrtps_cpp::TypeSupport<rosidl_typesupport_introspection_c__MessageMembers>::deserializeROSmessage(eprosima::fastcdr::Cdr&, rosidl_typesupport_introspection_c__MessageMembers const*, void*, bool)
    () from /home/jackie/code/ros2_ws/install/lib/librmw_fastrtps_cpp.so
#5  0x00007ffff6490155 in rmw_fastrtps_cpp::TypeSupport<rosidl_typesupport_introspection_c__MessageMembers>::deserializeROSmessage(rmw_fastrtps_cpp::Buffer const*, void*) ()
```